### PR TITLE
[W5.4][W14-A4] Jiang Chunhui

### PR DESCRIPTION
**Pull request by Adoby7**

### CODE REVIEW

File: doc/UserGuide.md [hunk 1]
```diff
@@ -81,6 +81,22 @@ Examples:
...
+> 
+> Put a `p` before the phone / email / address prefixes to mark it as `private`. `private` details can only
+> be seen using the `viewall` command.
+> 
+> Persons can have any number of tags (including 0)
+
+Examples: 
+* `update John Doe p/2222222 e/updated@gmail.com a/John street, block 123, #01-01`
+* `update Betsy Crowe pp/13579 e/betsycrowe@gmail.com pa/Newgate Prison t/criminal t/changed`
+
```
Comment (alvinlyj) [inline doc/UserGuide.md]: Have clearer instructions on how Update command works. 

For example does it update by name or index?

File: src/seedu/addressbook/commands/UpdateCommand.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+            return new CommandResult(String.format(MESSAGE_PERSON_NOT_FOUND, toUpdate.getName()));
+        }
+
+        try {
+            addressBook.addPerson(toUpdate);
+            return new CommandResult(String.format(MESSAGE_SUCCESS, toUpdate));
+        } catch (UniquePersonList.DuplicatePersonException dpe) {
+            return new CommandResult(MESSAGE_DUPLICATE_PERSON);
+        }
+    }
```
Comment (alvinlyj) [inline src/seedu/addressbook/commands/UpdateCommand.java]: Perhaps you can try to refactor this section to improve coding style and readability.

File: src/seedu/addressbook/parser/Parser.java [hunk 3]
```diff
@@ -106,32 +113,51 @@ public Command parseCommand(String userInput) {
...
-    private Command prepareAdd(String args) {
+    private Command prepareAddOrUpdate(boolean isAddCommand, String args) {
         final Matcher matcher = PERSON_DATA_ARGS_FORMAT.matcher(args.trim());
         // Validate arg string format
-        if (!matcher.matches()) {
+        if (!matcher.matches() && isAddCommand) {
             return new IncorrectCommand(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddCommand.MESSAGE_USAGE));
+        } else if (!matcher.matches() && !isAddCommand) {
+            return new IncorrectCommand(String.format(MESSAGE_INVALID_COMMAND_FORMAT, UpdateCommand.MESSAGE_USAGE));
         }
```
Comment (alvinlyj) [inline src/seedu/addressbook/parser/Parser.java]: Try to refactor your 'if' conditions.

File: test/java/seedu/addressbook/commands/UpdateCommandTest.java [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+        AddCommand addCommand = new AddCommand(p);
+        UpdateCommand updateCommand = new UpdateCommand(p);
+        addCommand.setData(book, EMPTY_PERSON_LIST);
+        addCommand.execute();
+        updateCommand.setData(book, EMPTY_PERSON_LIST);
+        CommandResult result = updateCommand.execute();
+
+        assertEquals(UpdateCommand.MESSAGE_UNNECESSARY_UPDATE, result.feedbackToUser);
+    }
+}
```
Comment (alvinlyj) [inline test/java/seedu/addressbook/commands/UpdateCommandTest.java]: Good testing done here, checking for many corner cases.

**Comment by alvinlyj** [comment]
@Adoby7 
Some comments added. Please acknowledge & close the PR after reading comments.
